### DISCUSSION TITLE
Snapdragon X Elite / Plus + NPU compatibility (megathread)

### DISCUSSION THREAD

**Discussion by magicAUS** [lllyasviel/Fooocus#3042] [General]
I have a Surface Pro 11th gen arriving in a few weeks. Moreso a query for the devs (@mashb1t ?)  - but will the current version Fooocus run on it, or should I not bother installing it until future versions?

(I'll post results on here if / when it arrives and has any success).

**Comment by andrewbabbittdev** [comment]
Will likely work under the emulation layer, but you will likely be limited to CPU generation which I expect would be pretty slow. I would not expect good performance out of it until someone adds support for NPUs

**Reply by mashb1t** [reply]
^ @magicAUS most likely this

**Reply by ErroneousBosch** [reply]
It looks like pyTorch has an NPU Package: https://pypi.org/project/torch-npu/

I am not enough of a Python person to know how it works.

**Reply by magicAUS** [reply]
> It looks like pyTorch has an NPU Package: https://pypi.org/project/torch-npu/
> 
> I am not enough of a Python person to know how it works.

Thanks for the link. Yeah, nor am I. I have a good PC setup, so will stick with that for now until NPU is supported in a build. Moreso just curious to compare the NPU performance against decent GPUs.

**Reply by ErroneousBosch** [reply]
I am pleased with how good Fooocus' performance is TBH. My modest 12gb 3060 on a basic prompt is at <1sec per iteration. I'm sure when NPUs come along they will be a boost.
